src/payment/checkout/Checkout.jsx

```diff
@@ -139,6 +143,10 @@ class Checkout extends React.Component {
     }
 
     const basketClassName = 'basket-section';
+
+    const loadingStripe = loading || !options.clientSecret;
+    const loadingCyberSource = loading && !enableStripePaymentProcessor;
```

Comment:
When `loadingCyberSource` is true, `loadingStripe` will always necessarily be true as well.  Seems like something is not quite accurate here?

It also looks like you set these consts, but only use them or'd together (line 179).  If what you care about on 179 is just `(loading && !enableStripePaymentProcessor) || !options.clientSecret`, you could just set a single const to that, and name it more appropriately (I'm not sure what it's doing, is it displaying a loading state?  if so e.g., `shouldDisplayLoadingState = ...`).

Reply:
That's a good point! I suppose the JS `||` will also never reach the right part of the statement if `loading` is truthy. 
We don't have a loading state of the billing and credit card fields for CyberSource, but for Stripe, at least locally I am finding that it takes a while to get the client secret ID (required for loading the credit card fields), so I added this in, and thought I should have it for CyberSource as well, hence the 2 separate variables. I will go ahead with your suggestion, thanks Colin! 

One issue is that the page could be done loading, have the client secret, but the Stripe fields are still not loaded.

Reply:
These are the states that currently affect this - 
CyberSource and Stripe: `loading`
Stripe: `enableStripePaymentProcessor`, `options.clientSecret`

But not having any or both of the stripe props above does not mean it's not a Stripe payment necessarily, but it's fetching this data. So `!enableStripePaymentProcessor` and `!options.clientSecret` does not mean it should be a CyberSource payment or it could mean it should!

Also, for `loading && !enableStripePaymentProcessor) || !options.clientSecret` if `enableStripePaymentProcessor` is true, and for some reason there no clientSecret (but loading is done), it would show the loading form instead of some error on obtaining the client secret.

Reply:
One important distinction to make is that `loading` is fetching the basket data, and is not related to the `clientSecretId` fetching. One TODO is to rename to `loadingBasket` to be more clear.